### ISSUE TITLE
RespondToChallenge optional branch can be skipped by bad actor

### ISSUE THREAD

**Issue by area** [JoinColony/colonyNetwork#410]
There is an 'optional' branch in `respondToChallenge`, which is executed if the submitter is proving that a reputation existed in `lastAgreeState`, and increments `challengeStepCompleted` an extra `1`. In certain circumstances (the reputation exists, and is going to become zero after the reputation update under dispute is applied), a bad actor can respond on behalf of a good actor, skipping this branch (even though the good actor could prove it), but otherwise successfully completing `respondToChallenge`, causing the good submission to be eliminated incorrectly.

**Comment by area** [comment]
After #506 is merged, I *think* that [this line](https://github.com/JoinColony/colonyNetwork/blob/691cb5d/contracts/ReputationMiningCycleRespond.sol#L359) is no longer needed - because `checkAdjacentReputation` will fail if they claim that a new reputation should be inserted when it already exists.

Given that these optional additions are the source of this issue, I wonder if there's a more elegant solution than the one I had in mind (which involves allowing `reputationMiningCycle` to be called multiple times, and recording the score the 'best' proof achieved).
